packages/asset/contracts/Asset.sol

```diff
@@ -385,6 +311,36 @@ contract Asset is
         return ERC1155URIStorageUpgradeable.uri(tokenId);
     }
 
+    function getTokenIdByMetadataHash(
+        string memory metadataHash
+    ) public view returns (uint256) {
+        return hashUsed[metadataHash];
+    }
+
+    function _setMetadataHash(
+        uint256 tokenId,
+        string memory metadataHash
+    ) internal onlyRole(MINTER_ROLE) {
+        if (hashUsed[metadataHash] != 0) {
+            require(
+                hashUsed[metadataHash] == tokenId,
+                "metadata hash mismatch for tokenId"
+            );
+        } else {
+            hashUsed[metadataHash] = tokenId;
+            _setURI(tokenId, metadataHash);
+        }
+    }
+
+    function getIncrementedCreatorNonce(
+        address creator
+    ) public onlyRole(MINTER_ROLE) returns (uint16) {
+        unchecked {
+            creatorNonces[creator]++;
+        }
+        return creatorNonces[creator];
+    }
```

Comment:
creatorNonces[creator] is never updated, uhhhh

Reply:
The below code is incrementing the nonce @mvanmeerbeck 

```
creatorNonces[creator]++;
```

Reply:
a "getter" is incrementing a var, i got tricked